### PR TITLE
Image Overlap issue

### PR DESCRIPTION
**Pull request by georgemoralis**
FIrst step:

Added logging to see why we reach this : 

UNREACHABLE_MSG("Encountered unresolvable image overlap with equal memory address.");

If you can provide me the log : 
known games : 
https://github.com/shadps4-compatibility/shadps4-game-compatibility/issues?q=is%3Aissue%20state%3Aopen%20Encountered%20unresolvable%20image%20overlap%20with%20equal%20memory%20address.

### CODE REVIEW

**Comment by georgemoralis** [comment]
Red dead redemption seems to have fewer issues

File: src/video_core/texture_cache/texture_cache.cpp [hunk 1]
```diff
@@ -358,7 +365,11 @@ std::tuple<ImageId, int, int> TextureCache::ResolveOverlap(const ImageInfo& imag
                   vk::to_string(image_info.pixel_format), int(image_info.type),
                   image_info.num_samples, int(image_info.tile_mode));
 
-        UNREACHABLE_MSG("Encountered unresolvable image overlap with equal memory address.");
```
Comment (StevenMiller123) [inline src/video_core/texture_cache/texture_cache.cpp:349]: I don't think removing the unreachable is the desired solution to this edge case

**Comment by StevenMiller123** [comment]
I've got a handful of games I can test this with, though I can't start testing for around 3 and a half more hours.

File: src/video_core/texture_cache/texture_cache.cpp (hunk 1 above)
Comment (georgemoralis) [reply src/video_core/texture_cache/texture_cache.cpp:349]: it is not removed just moved as a LOG after ... atm i tried to fix the overlaping of same images with different size

**Comment by diogoaduartereis** [comment]
Dying Light now only produces a black screen. The menu work fine though. And it goes in-game with just a black screen with UI.
[CUSA02010.log](https://github.com/user-attachments/files/25395525/CUSA02010.log)

Same with PT: 
[CUSA01127.log](https://github.com/user-attachments/files/25395567/CUSA01127.log)

Infamous Second Son crashes on startup:
[CUSA00223.log](https://github.com/user-attachments/files/25395598/CUSA00223.log)

Infamous First Light Crashes on startup:
[CUSA00897.log](https://github.com/user-attachments/files/25395622/CUSA00897.log)

**Comment by diogoaduartereis** [comment]
Latest Commit breaks Killzone shadow fall:
[CUSA00191.log](https://github.com/user-attachments/files/25397003/CUSA00191.log)

**Comment by diogoaduartereis** [comment]
image
Shadows are messed up I think. But there is no longer the tiny image on the corner of the screen like before on windows

**Comment by georgemoralis** [comment]
no glory for this try
